examples/deployment/kubernetes/deploy.sh

```diff
@@ -3,54 +3,69 @@
 # - Cluster has already been created & configured using the create.sh script
 # - Go 1.9 is installed
 
+function checkEnv() {
+  if [ -z ${PROJECT_NAME+x} ] ||
+     [ -z ${CLUSTER_NAME+x} ] ||
+     [ -z ${MASTER_ZONE+x} ] ||
+     [ -z ${CONFIGMAP+x} ]; then
+    echo "You must either pass an argument which is a config file, or set all the required environment variables" >&2
+    exit 1
+  fi
+}
+
 set -e
 
 DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
-source ${DIR}/config.sh
+if [ $# -eq 1 ]; then
+  source $1
+else
+  checkEnv
+fi
+
 
 export LOG_URL=TODO
 export MAP_URL=TODO
-export TAG=$(git rev-parse HEAD)
+export IMAGE_TAG=${IMAGE_TAG:-$(git rev-parse HEAD)}
```

Comment:
nit: The default `IMAGE_TAG` could be misleading, since the user may have modified but not committed some files which will subsequently get built into the image.

Reply:
I've made it append a `-dirty` string to the hash if that's the case.